xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp

```diff
@@ -230,16 +230,27 @@ void CVideoPlayerAudio::Process()
   m_audioStats.Start();
   m_disconAdjustCounter = 0;
 
-  // Only enable "learning" if advancedsettings m_maxPassthroughOffSyncDuration
-  // not exists or has it's default 10 ms value, otherwise use advancedsettings value
+  const float fps = m_processInfo.GetVideoFps();
+
```

Comment:
We might not know video fps on stream start and there is variable frame rate. Take care here.

Reply:
Good point, maybe check for `fps > 10.0` and if not set 25.0 fps as default?

Reply:
Key question rather is: Would it harm if we set it to 60 ms - always?

Reply:
or better: 64 ms - which is two AC3 packages? (and ~ 1.5 (1000.0/23.7658)?)

Reply:
Reasoning: If we use a 60 fps video - with AC3 soundtrack. The package time for one AC3 packet is 32 ms ... we could not change anything there to be more exact ... so two sounds okayish for passthrough?

Reply:
But some users still want use high values:

> On my Shield I need to use a high maxpassthroughoffsyncduration value or I will get lots of video stutters. Also some people are more sensitive to micro-stutters than others, there's a chance that it happened but you just didn't notice them. For me, I notice stutters very easily even when there's not a lot of movement.
> 
> If you need to make adjustments to the learning algorithm, that's fine as long as you keep the ability to set the advancedsetting up to 100 ms, because I definitely need it.

https://forum.kodi.tv/showthread.php?tid=377117&pid=3198214#pid3198214

Reply:
Probably for videos at 59.94 fps AE cannot make corrections if error is bigger than 33 ms (2 * frame time). Then 64 ms may not work for all videos.

Reply:
I wanted to keep the advancedsetting in place - users want to shoot their foot :-).

My point was, that when we have an AC3 soundtrack, pack that IEC wise - each packet that we add is 32 ms (byte payload vs. samplerate and internal audio 32 ms). So for AC3 it basically makes limited sense in the passthrough (resample for AC3 and  reecndoing is another story) to try to be more exact as 32 ms. And as we cannot split IEC packages, most likely 2 packages off is okayish - so that when we are "nearly 3 off" - we could ditch one, etc.

For high framerate videos the PTS control possibility for the video is much more fine granularly possible - while audio still is limited to 32 ms when soundtrack is AC3.

I leave the design choice to you - you have proven in the past to choose proper values ;-) ^^ that's just my reasoning.